### PR TITLE
[SPARK-7760] add /json back into master & worker pages; add test

### PR DESCRIPTION
**Pull request by squito**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala [hunk 1]
```diff
@@ -53,7 +55,8 @@ class LocalSparkCluster(
       .set("spark.shuffle.service.enabled", "false")
 
     /* Start the Master */
-    val (masterSystem, masterPort, _, _) = Master.startSystemAndActor(localHostname, 0, 0, _conf)
+    val (masterSystem, masterPort, webUi, _) = Master.startSystemAndActor(localHostname, 0, 0, _conf)
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala]: `s/webUi/webUiPort`

File: core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala [hunk 2]
```diff
@@ -157,4 +161,27 @@ class MasterSuite extends FunSuite with Matchers {
     CustomRecoveryModeFactory.instantiationAttempts should be > instantiationAttempts
   }
 
+  test("Master & worker web ui available") {
+    implicit val formats = org.json4s.DefaultFormats
+    val conf = new SparkConf()
+    val localCluster = new LocalSparkCluster(2, 2, 512, conf)
+    localCluster.start()
+    try {
+      eventually(timeout(2 seconds), interval(100 milliseconds)) {
```
Comment (vanzin) [inline core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala]: Is 2 seconds enough? Does `start()` guarantee that the WebUI is up and running when it returns?

I'm wary of some really slow build machine triggering this timeout.

**Comment by vanzin** [comment]
LGTM aside from a couple of minor things. Seems like there are some scalastyle issues though.

**Comment by squito** [comment]
So actually the last commit https://github.com/squito/spark/commit/f32948d5407033519d2ed140d8a585f538c0a1ea will be unnecessary once https://github.com/apache/spark/pull/6273 is merged.  I will undo it once that is merged, and then this will be ready for merging I think

**Comment by squito** [comment]
#6273 has been merged, and I updated this to reflect that, so I think this is now ready to go (assuming tests pass)

**Comment by squito** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala [hunk 3]
```diff
@@ -43,6 +43,8 @@ class LocalSparkCluster(
   private val localHostname = Utils.localHostName()
   private val masterActorSystems = ArrayBuffer[ActorSystem]()
   private val workerActorSystems = ArrayBuffer[ActorSystem]()
+  // exposed for testing
+  var masterWebUIPort = -1
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala]: Minor nit: we could use the `@VisibleForTesting` annotation here, but not a big deal (the codebase is a bit inconsistent about this).

**Comment by JoshRosen** [comment]
This looks good to me. The original JsonProtocolSuite tests for the actual contents of the responses weren't touched by the REST API patch, so I think that this test of the endpoints themselves should be sufficient to keep this from breaking in the future.  I'm going to merge this into master and branch-1.4.

**Pull request closed without merging**